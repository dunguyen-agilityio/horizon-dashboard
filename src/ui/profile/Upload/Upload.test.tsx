import { render, screen } from '@testing-library/react';

import UploadFile from '.';

describe('UploadFile ui', () => {
  it('should match snapshot', () => {
    const { container } = render(<UploadFile />);
    expect(container).toMatchSnapshot();
  });

  it('should render the UploadFile component correctly', () => {
    render(<UploadFile />);

    const uploadTitle = screen.getByText('Upload Files');
    expect(uploadTitle).toBeInTheDocument();

    const allowedFilesText = screen.getByText(
      'PNG, JPG and GIF files are allowed',
    );
    expect(allowedFilesText).toBeInTheDocument();

    const profileTitle = screen.getByText('Complete your profile');
    expect(profileTitle).toBeInTheDocument();

    const descriptionText = screen.getByText(
      'Stay on the pulse of distributed projects with an anline whiteboard to plan, coordinate and discuss',
    );
    expect(descriptionText).toBeInTheDocument();

    const publishButton = screen.getByRole('button', { name: 'Publish now' });
    expect(publishButton).toBeInTheDocument();
  });
});
