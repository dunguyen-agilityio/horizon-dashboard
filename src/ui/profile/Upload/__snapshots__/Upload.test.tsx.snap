// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadFile ui should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col sm:flex-row items-center gap-9 p-7 bg-white rounded-md dark:bg-indigo sm:h-[365px]"
  >
    <div>
      <label
        class="flex flex-col gap-6 justify-center items-center bg-grayest dark:bg-indigo-light w-[268px] h-[311px] rounded-md border-dashed border-2 border-amber-450 dark:border-secondary"
      >
        <input
          class="hidden"
          type="file"
        />
        <i
          class="h-6 flex justify-center items-center hover:cursor-pointer w-12 h-14fill-indigo-light fill-blue-450 dark:fill-white"
        >
          <svg
            height="56"
            viewBox="0 0 48 56"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.3334 42.3333H30.6668C32.5001 42.3333 34.0001 40.8333 34.0001 39V22.3333H39.3001C42.2668 22.3333 43.7668 18.7333 41.6668 16.6333L26.3668 1.33334C25.0668 0.0333374 22.9668 0.0333374 21.6668 1.33334L6.36675 16.6333C4.26675 18.7333 5.73341 22.3333 8.70008 22.3333H14.0001V39C14.0001 40.8333 15.5001 42.3333 17.3334 42.3333ZM4.00008 49H44.0001C45.8334 49 47.3334 50.5 47.3334 52.3333C47.3334 54.1667 45.8334 55.6667 44.0001 55.6667H4.00008C2.16675 55.6667 0.666748 54.1667 0.666748 52.3333C0.666748 50.5 2.16675 49 4.00008 49Z"
              fill="dark:white"
            />
          </svg>
        </i>
        <div>
          <p
            class="text-blue-450 dark:text-white text-blue-450 font-bold text-[20px] text-center"
          >
            Upload Files
          </p>
          <p
            class="text-primary dark:text-white font-normal text-secondary dark:!text-secondary text-[12px] text-center"
          >
            PNG, JPG and GIF files are allowed
          </p>
        </div>
      </label>
    </div>
    <div
      class="flex flex-col justify-around text-center sm:text-start"
    >
      <div>
        <p
          class="text-primary dark:text-white font-bold text-[24px]"
        >
          Complete your profile
        </p>
        <p
          class="text-primary dark:text-white text-medium font-normal text-secondary dark:!text-secondary"
        >
          Stay on the pulse of distributed projects with an anline whiteboard to plan, coordinate and discuss
        </p>
      </div>
      <div
        class="flex justify-center sm:justify-start mt-4"
      >
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-450 text-white w-1/2 dark:bg-purple-750"
          type="button"
        >
          Publish now
        </button>
      </div>
    </div>
  </div>
</div>
`;
